yellow-paper/docs/cryptography/performance-targets.md

```diff
@@ -0,0 +1,181 @@
+# Honk targets and win conditions
+
+## Introduction & context
+
+Aztec's cryptography tech stack and its associated implementation is an open-ended project with potential for many enhancements, optimisations and scope-creep.
+
+This document is designed to definitively answer the following questions:
+
+1. What are the metrics we care about when measuring our cryptography components?
+3. What are minimum satisfiable values for these metrics?
+4. What are the aspirational values for these metrics?
+
+# Important Metrics
+
+The following is a list of the relevant properties that affect the performance of the Aztec network:
+
+* Size of a user transaction (in kb)
+* Time to generate a user transaction proof
+* Memory required to generate a user transaction proof
+* Time to generate an Aztec Virtual Machine proof
+* Memory required to generate an Aztec Virtual Machine proof
+* Time to compute a 2x2 rollup proof
+* Memory required to compute a 2x2 rollup proof
+
+<!-- We can break these properties down into metrics linked to specitic cryptographic components:
+
+* Size of Goblin Plonk proofs
+* Size of Honk proofs
+* Honk prover time
+* Goblin Plonk prover time
+* Protogalaxy recursion -->
+
+ "MVP" = minimum standards that we can go to main-net with.
+
+Note: gb = gigabytes (not gigabits, gigibits or gigibytes)
+
+| metric | how to measure | MVP (10tps) | ideal (100tps) |
+| --- | --- | --- | --- |
+| proof size | total size of a user tx incl. goblin plonk proofs | 80kb | 8kb |
+| prover time | A baseline "medium complexity" transaction (in web browser). Full description further down | 1 min | 10 seconds |
+| verifier time | how long does it take the verifier to check a proof (incl. grumpkin IPA MSMs) | 20ms | 1ms |
+| client memory consumption | fold 2^19 circuits into an accumulator an arbitrary number of times | 4gb | 1gb |
+| size of the kernel circuit | number of gates | 2^17 | 2^15 |
+| Aztec Virtual Machine prover time | 1 million VM step circuit | 60 seconds | 6 seconds |
+| Aztec Virtual Machine memory consumption | 1 million VM step circuit | 128gb | 16gb |
+| 2x2 rollup proving time | 1 2x2 rollup proof | 7.4 seconds | 0.74 seconds |
+| 2x2 rollup memory consumption | 1 2x2 rollup proof | 128gb | 16gb |
+
+To come up with the above estimates, we are targetting 10 transactions per second for the MVP and 100 tps for the "ideal" case. We are assuming both block producers and rollup Provers have access to 128-core machines with 128gb of RAM. Additionally, we assume that the various process required to produce a block consume the following: 
```

Comment:
@zac-williamson currently we have been assuming 32 -64 cores max as the perfomance benefit drops off after that.

See spreadsheet here: https://docs.google.com/spreadsheets/d/1cBBZZ_dyD0tiUmAdjoGbnLrk2H0oSmOgYubtZy9JTRU/edit#gid=1562975724

Secondly, ideally block producers (sequencers) can be run on 16 core 32gb machines as they will not be producing proofs.